landingai/pipeline/frameset.py

```diff
@@ -41,18 +38,36 @@ class PredictionList(List[ClassificationPrediction]):
     False
     """
 
+    def __init__(self, *args: Any) -> None:
+        # TODO: This is a hack to support OCR predictions. We should probably have different PredictionList
+        #  classes for each type of prediction, so we don't have to do conditional checks everywhere and `cast`
+        # or "type: ignore".
+        super().__init__(*args)
+        for p in self:
+            if not isinstance(p, type(self[0])):
+                raise ValueError("All elements should be of the same type")
+
+    @property
+    def inner_type(self) -> str:
```

Comment:
But it has to be used by `landingai.pipeline.postprocessing.get_class_counts` , otherwise that function is broken. In general, I prefer not to use private methods outside of a class, but I would understand if you say so that no user of the SDK would depend on it.

What do you think?

Reply:
Ideally, `get_class_counts` should be testing for `PredictionList` subclass type (`if isinstance(frame.predictions, OCRPredList): whatever`). Now it is not possible to do it because of the bad design choices made in the past, that uses the same `PredictionList` class for every type of prediction...

I would prefer having this as a `_` private attribute (to mark to end users that they shouldn't use this), and breaking the encapsulation by accessing the private private attribute on `get_class_counts`.